Applications/MaterialAssist/Client/MaterialAssistClientBoards.cs

```diff
@@ -0,0 +1,196 @@
+        private const string _BoardCode = "boardCode";
+        private const string _BoardTypeCode = "boardTypeCode";
+        private const string _StorageLocation = "storageLocation";
+        private const string _DeleteFromInventory = "deleteFromInventory";
+        private const string _Length = "length";
+        private const string _Width = "width";
+        private const string _Comments = "comments";
+        private const string _RemovalType = "removalType";
+        private const string _Quantity = "quantity";
+
+        #endregion Constants
+
+        #region Read
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BoardEntity>> GetBoardEntities(int take, int skip = 0)
+        {
+            var url = $"{_BaseRoute}?take={take}&skip={skip}";
+
+            return await RequestEnumerable<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<BoardEntity> GetBoardEntityById(string id)
+        {
+            var url = $"{_BaseRoute}?{_Id}={Uri.EscapeDataString(id)}";
+
+            return await RequestObject<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BoardEntity>> GetBoardEntitiesByIds(IEnumerable<string> ids)
+        {
+            var url = $"{_BaseRoute}";
+
+            var boardCodes = new StringBuilder("?");
+            boardCodes.Append(string.Join("&", ids.Select(id => $"{_Id}={Uri.EscapeDataString(id)}")));
+
+            url += boardCodes;
+
+            return await RequestEnumerable<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BoardEntity>> GetBoardEntitiesByBoardCode(string boardCode)
+        {
+            var url = $"{_BaseRoute}?{_BoardCode}={Uri.EscapeDataString(boardCode)}";
+
+            return await RequestEnumerable<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BoardEntity>> GetBoardEntitiesByBoardCodes(IEnumerable<string> boardCodes)
+        {
+            var url = $"{_BaseRoute}";
+
+            var appendedBoardCodes = new StringBuilder("?");
+            appendedBoardCodes.Append(string.Join("&", boardCodes.Select(boardCode => $"{_BoardCode}={Uri.EscapeDataString(boardCode)}")));
+
+            url += boardCodes;
+
+            return await RequestEnumerable<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<StorageLocation>> GetStorageLocations()
+        {
+            var url = $"{_BaseRoute}/storageLocations";
+
+            return await RequestEnumerable<StorageLocation>(new Uri(url, UriKind.Relative));
+        }
+
+        #endregion Read
+
+        #region Create
+
+        /// <inheritdoc />
+        public async Task<BoardEntity> CreateBoardEntity(string boardTypeCode, BoardEntity boardEntity)
```

Comment:
Doesn't have the BoardEntity a property boardtypeCode?

Reply:
BoardEntity has a property of BoardType and BoardType has a property of BoardTypeCode.
As we do not allow to do both at the same time (the BoardType has to be created before the instance) we don´t have it in this case

Reply:
removed